### PR TITLE
Feature/add rediscoverte

### PR DESCRIPTION
**Pull request by nikhil**
Adds the modules needed for the rediscoverte pipeline
- rediscoverte
- Salmon (index and quant)

### CODE REVIEW

File: modules/msk/rediscoverte/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
```
Comment (johnoooh) [inline modules/msk/rediscoverte/tests/main.nf.test]: Remove TODO

File: modules/msk/rediscoverte/environment.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
+name: "rediscoverte"
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - "YOUR-TOOL-HERE"
```
Comment (johnoooh) [inline modules/msk/rediscoverte/environment.yml]: Add your  tool

File: modules/msk/salmon/quant/main.nf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    salmon quant \\
+        --index ${index} \\
+        --seqBias \\
+        --gcBias \\
+        --libType A \\
+        --unmatedReads ${reads} \\
+        --validateMappings \\
+        -o ${prefix}.salmon.quant \\
+        --threads ${threads}
```
Comment (johnoooh) [inline modules/msk/salmon/quant/main.nf:37]: Do we need to add $args to the end of this?

File: modules/msk/salmon/index/main.nf [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def threads = task.cpus * 2
+    """
+    salmon \\
+        index \\
+        --threads $threads \\
+        -t $transcriptome_fasta \\
+        -i salmon
```
Comment (johnoooh) [inline modules/msk/salmon/index/main.nf:28]: $args here maybe?

**Review by johnoooh — CHANGES_REQUESTED** [review]
Some tools may need ${args} added to them.  What do you think Nikhil?

File: modules/msk/salmon/quant/main.nf (hunk 3 above)
Comment (nikhil) [reply modules/msk/salmon/quant/main.nf:37]: we should!

File: modules/msk/salmon/index/main.nf (hunk 4 above)
Comment (nikhil) [reply modules/msk/salmon/index/main.nf:28]: yup!

**Review by johnoooh — APPROVED** [review]
Looks good

**Pull request merged**